corptools/src/pages/Contacts.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import {
+  ButtonToolbar,
+  ButtonGroup,
+  Glyphicon,
+  MenuItem,
+  SplitButton,
+  Table,
+} from "react-bootstrap";
+
+import { Panel } from "react-bootstrap";
+import { useQuery } from "react-query";
+import { loadWallet } from "../apis/Character";
+import { PanelLoader } from "../components/PanelLoader";
+import { Button } from "react-bootstrap";
+import { useTable, usePagination } from "react-table";
+
+const Checkbox = ({ active }) => {
+  return (
+    <Button className={active ? "btn-success" : "btn-default"}>
+      <Glyphicon glyph={active ? "ok" : "remove"} />
+    </Button>
+  );
+};
+
+const CharContacts = ({ character_id }) => {
+  const { isLoading, error, data } = useQuery(
+    ["wallet", character_id],
+    () => loadContacts(character_id),
+    {
+      initialData: [],
+    }
+  );
+
+  const columns = React.useMemo(
```

Comment:
Are you modifying the columns anywhere? If they're not dynamic, just define a static array outside the component.

Reply:
ah ok saw some others and it's just a repeat pattern. nbd.